packages/components/src/Notes.js

```diff
@@ -1,15 +1,31 @@
-import React, { useEffect } from 'react'
+import React, { useEffect, useRef } from 'react'
 import { useDeck } from './context'
+import { useStep } from './useStep'
 
 export const Notes = props => {
   const context = useDeck()
+  if (!context || !context.register) {
+    // The slide was rendered as a preview by the <Presenter> component.
+    return false
+  }
+
+  const index = useRef(context.index).current
+  const step = useStep()
```

Comment:
In my experience, the `context` may not be for the slide that this `<Notes>` component was rendered for. So we need to cache the first `context.index` we get our hands on. For some reason, we can't do the same with `context.step`, as that results in race conditions.